shell/src/main/java/org/apache/accumulo/shell/Shell.java

```diff
@@ -189,6 +189,16 @@
  */
 @AutoService(KeywordExecutable.class)
 public class Shell extends ShellOptions implements KeywordExecutable {
+
+  public static interface Authenticator {
+    default boolean authenticateUser(AccumuloClient client, String principal,
+        AuthenticationToken token) throws AccumuloException, AccumuloSecurityException {
+      return client.securityOperations().authenticateUser(principal, token);
+    }
+  }
+
+  public static Authenticator AUTHENTICATOR = new Authenticator() {};
```

Comment:
Does this exist so that a test can override a static method?

Reply:
Yes, FateCommandTest

Reply:
ok after looking at the PR that is what I suspected.  When I first saw `AUTHENTICATOR` I was wondering why it could not be replaced with a static method.

Reply:
Note: FateCommandTest is removed in 3.0, since the FateCommand is deprecated in 2.1 and removed in 3.0.

I think that making this static is a confusing way to implement this for testing, though. I think it would be better to use a protected method and create a subclass test shell in FateCommandTest.createShell that overrides this method for testing. That way, there's less public and less static stuff leaking out of the shell's API for users to shoot themselves in the foot with.

Reply:
I added commit 2db232fc79eaf640d43c81da4fba19dab7bcd0e7 to implement my suggestions. This commit can be rolled back out or tweaked, if needed.